### DISCUSSION TITLE
1+5 OOS 4.5.2 vpn hotspot no internet on client

### DISCUSSION THREAD

**Discussion by hojiao** [Mygod/VPNHotspot#321] [Q&A] [answered]
Dev, can you please have a look? No internet on client when vpn + wlan0
It can/may work when the system is fresh (just flashed OOS or when OOS rescue mechanism is triggered the system settings will reset to default or {new vpn that uses new interface like tun1, tun2, i guess})
Then after turn off and on hotspot / on and off airplane, it won't work anymore (need to try many many many times clear/reapply or other vpn hotspot bash script, even can spend few hours and get nothing)
But I realized if use network signal guru to switch mobile data band and quickly refresh on client (same page, example my ip page), it will display vpn ip may/not load finish the page, but definitely cannot go to other page (no internet, more like dns problem?)
Client already set ipv4 dns to 8.8.8.8, 8.8.4.4
Client can ping 8.8.8.8, google.com, but no internet in browser
Phone: OOS 4.5.2 (Nougat)
VPN Hotspot: 2.3.2 (Same results on every version)
Tried toggling the disable ipv6 and enable dhcp in app, setting upstream interface, no luck
[log1-working](https://forum.xda-developers.com/attachments/vpnhotspot-55653763-working-zip.5501271/) 
[log2-working](https://forum.xda-developers.com/attachments/vpnhotspot-1716699749-working-zip.5501275/) 
[log3-notworking](https://forum.xda-developers.com/attachments/vpnhotspot-1510774627-notworking-zip.5501273/)

**Comment by Mygod** [comment]
Use the latest version of the app and export logs from there.

**Comment by hojiao** [comment, marked as answer]
finally solved 
need to disable qti tether this 💩

**Reply by Mygod** [reply]
Interesting. How did you disable that?

**Reply by hojiao** [reply]
i disabled the system app com.qualcomm.qti.tetherservice and removed the libQtiTether.so in /vendor/lib and /vendor/lib64
